src/main/java/com/karankumar/bookproject/backend/goal/CalculateReadingGoal.java

```diff
@@ -49,7 +49,7 @@ public static int shouldHaveRead(int booksToReadThisYear) {
      * the year
      */
     public static int booksToReadFromStartOfYear(int booksToReadThisYear) {
-        return ((int) Math.ceil(booksToReadThisYear / WEEKS_IN_YEAR));
+        return (int)((float)booksToReadThisYear / WEEKS_IN_YEAR);
```

Comment:
I noticed is that this method shouldn't return an `int`. If the value of `booksToReadThisYear` is 26, it doesn't make much sense to return 0 -- we should return 0.5.

Could you please refactor this method to return a `double`? We should also make `shouldHaveRead()` return a `double`.

With `behindOrAheadSchedule()`, could you make the second argument a `double` and then ensure that when comparing the two parameters, you're handling the precision error problem.

Apologies for the increase in scope!

Reply:
Okay will do but is taking the number of books as numbers with decimals fine? Shouldn't they be integers?

Reply:
I know what you mean, but I think in this case it's fine. I think it's better to report a decimal back (e.g. 0.5 books need to be read on average per week from the start of the year if the reading goal is 26 books then to report 0)